File: solution/70. Climbing Stairs/70.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp2 = 0, dp1 = 1, dpi = 0;
        for (int i = 1; i <= n; ++i) {
            dpi = dp2 + dp1;
            dp2 = dp1;
            dp1 = dpi;
        }
        return dp1;
    }
};

File: solution/70. Climbing Stairs/70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp2 = dpi = 0
        dp1 = 1
        for i in range(n):
            dpi = dp2 + dp1
            dp2, dp1 = dp1, dpi
        return dpi
    
